### PR TITLE
Remove FE hacks for empty display names for column groups

### PR DESCRIPTION
**Pull request by ranquild**
Epic https://github.com/metabase/metabase/issues/47800
Closes https://github.com/metabase/metabase/issues/47811

### CODE REVIEW

File: src/metabase/lib/column_group.cljc [hunk 1]
```diff
@@ -72,9 +72,9 @@
         (lib.metadata.calculation/display-info query stage-number table))
       (when-let [card (some->> (:source-card stage) (lib.metadata/card query))]
         (lib.metadata.calculation/display-info query stage-number card))
-      ;; for multi-stage queries return an empty string (#30108)
+      ;; multi-stage queries (#30108)
       (when (next (:stages query))
-        {:display-name ""})
+        {:display-name "Summaries"})
```
Comment (romeovs) [inline src/metabase/lib/column_group.cljc:77]: Is this the corresponding stage always a summarization?

File: frontend/src/metabase/querying/fields/components/FieldPanel/utils.ts [hunk 2]
```diff
@@ -43,7 +41,7 @@ function getGroupsWithColumns(
 
     return {
       columnItems,
-      displayName: groupInfo.displayName || t`Question`,
```
Comment (romeovs) [inline frontend/src/metabase/querying/fields/components/FieldPanel/utils.ts:46]: AFAICT the goal of this PR is to not change any functionality and we're removing one `Question` label without replacing it with an equivalent on the back end.

It is not clear to me that this does not change functionality in this case.

**Review by romeovs — APPROVED** [review]
Code looks good but it's unclear if this actually leaves functionality untouched completely.

File: frontend/src/metabase/querying/fields/components/FieldPanel/utils.ts (hunk 2 above)
Comment (ranquild) [reply frontend/src/metabase/querying/fields/components/FieldPanel/utils.ts:46]: It replaces it with `Summaries` and makes things consistent.

File: src/metabase/lib/column_group.cljc (hunk 1 above)
Comment (ranquild) [reply src/metabase/lib/column_group.cljc:77]: You can't create multiple stage queries without summarization.

**Review by uladzimirdev — APPROVED** [review]

**Pull request merged**